### ISSUE TITLE
iotedge check does not detect journald logging as valid log rotation

### ISSUE THREAD

**Issue by mtrensch** [Azure/iotedge#6728]
## Expected Behavior
iotedge check should detect journald as valid log rotation

## Current Behavior
iotedge check reports a warning about missing log rotation:

```
$ sudo iotedge check
...
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.
...
```
## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. Configure docker/moby to use journald for logging:
```/etc/docker/daemon.json
{
  "log-driver": "journald"
}
```
2. Run iotedge check 

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```
Connectivity checks (aziot-identity-service)
--------------------------------------------
√ host can connect to and perform TLS handshake with iothub AMQP port - OK
√ host can connect to and perform TLS handshake with iothub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with iothub MQTT port - OK

Configuration checks
--------------------
√ aziot-edged configuration is well-formed - OK
√ configuration up-to-date with config.toml - OK
√ container engine is installed and functional - OK
√ configuration has correct URIs for daemon mgmt endpoint - OK
‼ aziot-edge package is up-to-date - Warning
    Installed IoT Edge daemon has version 1.2.5 but 1.4.2 is the latest stable version available.
    Please see https://aka.ms/iotedge-update-runtime for update instructions.
√ container time is close to host time - OK
‼ DNS server - Warning
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
‼ production readiness: container engine - Warning
    Device is not using a production-supported container engine (moby-engine).
    Please see https://aka.ms/iotedge-prod-checklist-moby for details.
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its /tmp/edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
‼ production readiness: Edge Hub's storage directory is persisted on the host filesystem - Warning
    The edgeHub module is not configured to persist its /tmp/edgeHub directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
√ Agent image is valid and can be pulled from upstream - OK

Connectivity checks
-------------------
√ container on the default network can connect to upstream  AMQP port - OK
√ container on the default network can connect to upstream HTTPS / WebSockets port - OK
√ container on the default network can connect to upstream MQTT port - OK
√ container on the IoT Edge module network can connect to upstream AMQP port - OK
√ container on the IoT Edge module network can connect to upstream HTTPS / WebSockets port - OK
√ container on the IoT Edge module network can connect to upstream MQTT port - OK
29 check(s) succeeded.
7 check(s) raised warnings. Re-run with --verbose for more details.
```
</details>

### Device Information
* OS: Linux (yocto build) 
* Architectures: amd64, arm32 and arm64 
* Container OS : Linux

### Runtime Versions
* aziot-edged [run `iotedge version`]: 1.2.5
* Docker/Moby [run `docker version`]:  20.10.11

## Additional information
I am no rust expert, but the following should help to make journald a valid log rotation mechanism:
```
diff -aurN a/edgelet/iotedge/src/check/checks/container_engine_logrotate.rs b/edgelet/iotedge/src/check/checks/container_engine_logrotate.rs
--- a/edgelet/iotedge/src/check/checks/container_engine_logrotate.rs    2022-09-26 08:51:22.022133573 +0200
+++ b/edgelet/iotedge/src/check/checks/container_engine_logrotate.rs    2022-09-26 12:38:23.828987471 +0200
@@ -55,6 +55,11 @@
             .context(MESSAGE)?;
         self.daemon_config = Some(daemon_config.clone());

+        let journald: Option<String> = Some("journald".to_owned());
+        if journald == daemon_config.log_driver {
+            return Ok(CheckResult::Ok);
+        }
+
         if daemon_config.log_driver.is_none() {
             return Ok(CheckResult::Warning(anyhow!(MESSAGE)));
         }
```

**Comment by raisalitch** [comment]
@mtrensch Can you try adding `max-size` and `max-file` options? https://learn.microsoft.com/en-us/azure/iot-edge/production-checklist?view=iotedge-2020-11#set-up-default-logging-driver

**Comment by mtrensch** [comment]
@raisalitch Those options are not valid for journald logging driver (see https://docs.docker.com/config/containers/logging/journald/#options).
Adding those to daemon.json results in docker not starting up.
```
dockerd[18916]: failed to start daemon: failed to set log opts: unknown log opt 'max-size' for journald log driver
systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: docker.service: Failed with result 'exit-code'.
systemd[1]: Failed to start Docker Application Container Engine.
```

**Comment by nlcamp** [comment]
@mtresch - I would encourage you to submit this as a pull request.

**Comment by mtrensch** [comment]
@nlcamp - PR submitted, but as c developer I cannot judge if this is "proper" rust ;)

**Comment by nlcamp** [comment]
@mtrensch - Thanks for this contribution. Your PR has been merged and should be included in the next iotedge 1.4.x release.
